browser/brave_cosmetic_resources_tab_helper.cc

```diff
@@ -0,0 +1,472 @@
+          "}"
+          "rule += prop;"
+        "});"
+        "rule += '}';"
+        "window.content_cosmetic.cosmeticStyleSheet.insertRule("
+          "`${rule}`, nextIndex);"
+        "window.content_cosmetic.allSelectorsToRules.set("
+          "selector, nextIndex);"
+        "nextIndex++;"
+      "};"
+    "};"
+    "if (!document.adoptedStyleSheets.includes("
+          "window.content_cosmetic.cosmeticStyleSheet)){"
+       "document.adoptedStyleSheets ="
+         "[window.content_cosmetic.cosmeticStyleSheet];"
+    "};"
+  "})();";
+
+std::vector<std::string>
+    BraveCosmeticResourcesTabHelper::vetted_search_engines_ = {
+  "duckduckgo",
+  "qwant",
+  "bing",
+  "startpage",
+  "google",
+  "yandex",
+  "ecosia"
+};
+
+// Doing that to prevent a lint error:
+// Static/global string variables are not permitted
+// You can create an object dynamically and never delete it by using a
+// Function-local static pointer or reference
+// (e.g., static const auto& impl = *new T(args...);).
+std::string* BraveCosmeticResourcesTabHelper::observing_script_ =
+    new std::string("");
+
+
+namespace {
+
+bool ShouldDoCosmeticFiltering(content::WebContents* contents,
+    const GURL& url) {
+  Profile* profile = Profile::FromBrowserContext(contents->GetBrowserContext());
+  auto* map = HostContentSettingsMapFactory::GetForProfile(profile);
+
+  return ::brave_shields::ShouldDoCosmeticFiltering(map, url);
+}
+
+std::string LoadDataResource(const int& id) {
+  std::string data_resource;
+
+  auto& resource_bundle = ui::ResourceBundle::GetSharedInstance();
+  if (resource_bundle.IsGzipped(id)) {
+    data_resource = resource_bundle.LoadDataResourceString(id);
+  } else {
+    data_resource = resource_bundle.GetRawDataResource(id).as_string();
+  }
+
+  return data_resource;
+}
+
+std::unique_ptr<base::ListValue> GetUrlCosmeticResourcesOnTaskRunner(
+    const std::string& url) {
+  auto result_list = std::make_unique<base::ListValue>();
+
+  base::Optional<base::Value> resources = g_brave_browser_process->
+      ad_block_service()->UrlCosmeticResources(url);
+
+  if (!resources || !resources->is_dict()) {
+    return result_list;
+  }
+
+  base::Optional<base::Value> regional_resources = g_brave_browser_process->
+      ad_block_regional_service_manager()->UrlCosmeticResources(url);
+
+  if (regional_resources && regional_resources->is_dict()) {
+    ::brave_shields::MergeResourcesInto(
+        std::move(*regional_resources), &*resources, /*force_hide=*/false);
+  }
+
+  base::Optional<base::Value> custom_resources = g_brave_browser_process->
+      ad_block_custom_filters_service()->UrlCosmeticResources(url);
+
+  if (custom_resources && custom_resources->is_dict()) {
+    ::brave_shields::MergeResourcesInto(
+        std::move(*custom_resources), &*resources, /*force_hide=*/true);
+  }
+
+  result_list->Append(std::move(*resources));
+
+  return result_list;
+}
+
+std::unique_ptr<base::ListValue> GetHiddenClassIdSelectorsOnTaskRunner(
```

Comment:
Both of these two functions shouldn't belong to the tab helper, rather should be somewhere close to the ad block services. For now we can put a TODO at least

Reply:
Related to that follow up https://github.com/brave/brave-browser/issues/13479